src/core/linter-rules/check-charset.js

```diff
@@ -1,35 +1,31 @@
 // @ts-check
 /**
- * Linter rule "check-charset".
- *
  * Checks whether the document has `<meta charset="utf-8">` properly.
  */
-import LinterRule from "../LinterRule.js";
-import { lang as defaultLang } from "../l10n.js";
+import { getIntlData, showWarning } from "../utils.js";
 
-const name = "check-charset";
-const meta = {
+const ruleName = "check-charset";
+export const name = "core/linter-rules/check-charset";
+
+const localizationStrings = {
   en: {
-    description: `Document must only contain one \`<meta>\` tag with charset set to 'utf-8'`,
-    howToFix: `Add this line in your document \`<head>\` section - \`<meta charset="utf-8">\` or set charset to "utf-8" if not set already.`,
+    msg: `Document must only contain one \`<meta>\` tag with charset set to 'utf-8'`,
+    hint: `Add this line in your document \`<head>\` section - \`<meta charset="utf-8">\` or set charset to "utf-8" if not set already.`,
   },
   zh: {
-    description: `文档只能包含一个 charset 属性为 utf-8 的 \`<meta>\` 标签`,
-    howToFix: `将此行添加到文档的 \`<head>\` 部分—— \`<meta charset="utf-8">\` 或将 charset 设置为 utf-8（如果尚未设置）。`,
+    msg: `文档只能包含一个 charset 属性为 utf-8 的 \`<meta>\` 标签`,
+    hint: `将此行添加到文档的 \`<head>\` 部分—— \`<meta charset="utf-8">\` 或将 charset 设置为 utf-8（如果尚未设置）。`,
   },
 };
+const l10n = getIntlData(localizationStrings);
 
-// Fall back to english, if language is missing
-const lang = defaultLang in meta ? defaultLang : "en";
+export function run(conf) {
+  if (!conf.lint?.[ruleName]) {
```

Comment:
As linter names have hyphens, we need to configure them as:
```js
var respecConfig = {
 lint: {
   "check-charset": true,
  },
};
```

maybe we can alias the config options so we can use:
```js
var respecConfig = {
 lint: {
   checkCharset: true,
  },
};
```

Reply:
yeah, argh... really screwed that up too.

Reply:
Will do that when all other linters have migrated to `run()` API.

Reply:
Filed https://github.com/w3c/respec/issues/3438 as a TODO